purchase_propagate_qty_mrp/models/purchase_order_line.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import _, models
+from odoo.exceptions import UserError
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = "purchase.order.line"
+
+    def _propagage_qty_to_moves(self):
+        kit_purchase_lines = self.env["purchase.order.line"].browse()
+        for line in self:
+            if line.state != "purchase":
+                continue
+            kit_line = line._propagate_qty_to_moves_mrp()
+            if kit_line:
+                kit_purchase_lines |= line
+        super(PurchaseOrderLine, self - kit_purchase_lines)._propagage_qty_to_moves()
+
+    def _propagate_qty_to_moves_mrp(self):
+        self.ensure_one()
+        bom = self.env["mrp.bom"].sudo()._bom_find(product=self.product_id)
```

Comment:
Yes _bom_find should be used but at first the moves should be checked. 
If there is no move with a bom line which linked to a bom for the purchased product than _bom_find should be used. 
This is the case for a kits in kit scenario then there is no bom for the main product. 
```suggestion
        boms = self.move_ids.bom_line_id.bom_id
        relevant_bom = False
        for bom in boms:
            if bom.type == "phantom" and (bom.product_id == self.product_id or bom.product_tmpl_id == self.product_id.product_tmpl_id and not bom.product_id):
                relevant_bom = bom
                break
        if not relevant_bom:
            relevant_bom = self.env["mrp.bom"].sudo()._bom_find(product=self.product_id, bom_type="phantom")
        bom = relevant_bom
```

Reply:
Thanks for your review. Using the `bom_line_id` on stock.move to find the related bom sounds much better.

Do you think using `_bom_find` is still necessary, because anyway no matching will be possible between the stock.move and the bom.line from the explode.

I will not include it yet.

Reply:
Yes it is still necessary for a kit in kit scenario. 
For example:
P-Kit-1 -> includes P1 and P2
P-Kit-2 -> includes P3 and P4
P-Kit-3 -> includes P-Kit-1 and P-Kit-2

The you will end update with moves for P1, P2, P3 and P4 no move will be link to the bom of P-Kit-3, the will be linked to P-Kit-1 bomlines and P-Kit-2. You will never find a bom for P-Kit-3.